selfservice/flow/recovery/error.go

```diff
@@ -69,8 +69,17 @@ func (s *ErrorHandler) WriteFlowError(
 	}
 
 	if e := new(flow.ExpiredError); errors.As(err, &e) {
+		strategy, err := s.d.RecoveryStrategies(r.Context()).Strategy(f.Active.String())
+		if err != nil {
+			strategy, err = s.d.GetActiveRecoveryStrategy(r.Context())
+			// Can't retry the recovery if no strategy has been set
+			if err != nil {
+				s.d.SelfServiceErrorManager().Forward(r.Context(), w, r, err)
+				return
+			}
+		}
```

Comment:
Is there no easier way to get this?

Reply:
Not really sure how, tbh.

We first need to check if the strategy of the "old" flow is still available. If it isn't, we fall back to the active strategy, and if that's not available we need to fail the flow.